tools/functional-tester/etcd-runner/watcher.go

```diff
@@ -0,0 +1,209 @@
+// Copyright 2016 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"math/rand"
+	"sync"
+	"time"
+
+	"github.com/coreos/etcd/clientv3"
+	"golang.org/x/time/rate"
+)
+
+func runWatcher(getClient getClientFunc, limit int) {
+	ctx := context.Background()
+	for round := 0; round < limit; round++ {
+		performWatchOnPrefixes(ctx, getClient, round)
+	}
+}
+
+func performWatchOnPrefixes(ctx context.Context, getClient getClientFunc, round int) {
```

Comment:
@xiang90 The othe issue which i feel with this code is that if we change etcd runner, to have flags for running time, noOfPrefixes, watcherPePrefix, reqRate the only way we can have it with the current design is to pass it as args (which will again make it complicated as will end up passing too many args to it) or to use these flags as global args which again will be a problem. Its just an example which i have mentioned which may or may not be applicable here but can happen in future for this or for other newly added etcd-runner feature. So probably making it exactly like etcdctl usage will be more appropriate as you had mentioned earlier.